### PR TITLE
Issue 2- Fix build for STM32L4 series

### PR DESCRIPTION
**Pull request by JeremySavonet**
STM32L4 series have a different power interface clock management since it's a
ultra-low energy series.
APB1 peripheral registers are named a bit differently.

### CODE REVIEW

**Comment by thirtytwobits** [comment]
👀 ...

**Comment by thirtytwobits** [comment]
Unfortunately I'm at a loss to actually verify this fix (I don't have representative hardware on hand). @JeremySavonet , would you mind providing a comment on how you tested this change? You have a pretty solid github history so I'm inclined to accept this change with a bit more assurance that it actually works. Thanks again for contributing.

**Comment by JeremySavonet** [comment]
@pavel-kirienko : Could you point me out where CS change need to be done. As I read the file some indent are like I do and others like you want.
Thanks in advance

**Comment by JeremySavonet** [comment]
@thirtytwobits: sure!
We have a custom board with a STM32L496 running ChibiOS 18.2.3.  On the board, there is also some sensors so that we can get data from them and push them on the CAN bus.
We ordered this product https://zubax.com/products/babel and then where able to properly see what we are pushing on the CAN bus.

File: driver/src/uc_stm32_can.cpp [hunk 1]
```diff
@@ -983,6 +983,27 @@ void CanDriver::initOnce()
      */
     {
         CriticalSectionLocker lock;
+#if defined(STM32L4xx)
```
Comment (thirtytwobits) [inline driver/src/uc_stm32_can.cpp]: I think @pavel-kirienko wants spaces between the # and if for these macros.

**Comment by dpslwk** [comment]
I think I have the correct indentation spacing here https://github.com/dpslwk/libuavcan_stm32/commit/c9cc7c7a05b648383ddad3030b32a981f368cdf9
@JeremySavonet happy for you to grab a patch from my fork if you want

I've also tested these changes on a NUCLEO-L443RC-P dev board with the same changes applied to the main libuavcan repo version of the driver, running just basic test under FreeRTOS & CubeMX

**Comment by JeremySavonet** [comment]
@dpslwk Thank you for the patch

**Comment by thirtytwobits** [comment]
For V1 I'd like to push vendors to pull UAVCAN drivers into their toolchains. For v0 I'm uncertain the level of support we can give for any specific driver. We may want to move the existing driver repos into a "UAVCAN commons" project that is clearly labeled as "unsupported commons - caveat emptor"

**Comment by pavel-kirienko** [comment]
> For V1 I'd like to push vendors to pull UAVCAN drivers into their toolchains.

I don't follow how does one integrate a ~~driver~~ platform interface (?) into a toolchain. Could you describe that on the object level?

**Comment by thirtytwobits** [comment]
Sorry. I didn't mean toolchain. I meant SDK.

**Comment by pavel-kirienko** [comment]
Ah, yes, indeed. I might have advised several users to do just that already so we seem aligned then. Platform interfaces are hard to generalize considering huge variability of hardware features and application constraints.

**Pull request merged**